src/libraries/Common/src/Interop/OSX/Interop.libc.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libc
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct AttrList
+        {
+            public ushort bitmapCount;
+            public ushort reserved;
+            public uint commonAttr;
+            public uint volAttr;
+            public uint dirAttr;
+            public uint fileAttr;
+            public uint forkAttr;
+
+            public const ushort ATTR_BIT_MAP_COUNT = 5;
+            public const uint ATTR_CMN_CRTIME = 0x00000200;
+        }
+
+        [DllImport(Libraries.libc, EntryPoint = "setattrlist", SetLastError = true)]
```

Comment:
From what I see, `Libraries.libc` is actually not used anywhere, so this is a first.
Probably this should move to SystemNative with a shim function and struct.

Reply:
Hi, do you mean like how [this](https://github.com/dotnet/runtime/blob/b0aa6f4928ae968024ca28f52b08a2353f87d0bc/src/libraries/Common/src/Interop/Unix/System.Native/Interop.LChflags.cs#L18) function is?
I can give it a go, but I haven't done that before.

Reply:
If so, would `SystemNative_OSX_setattrlist` be an appropriate name? Or should it be just `SystemNative_setattrlist`?

Reply:
And what `#if` could I use to conditionally compile it.

Reply:
>  do you mean like how this function is?

Yes. Lets wait for one of the `System.IO` owners to confirm they prefer the shim.

> If so, would SystemNative_OSX_setattrlist be an appropriate name? Or should it be just SystemNative_setattrlist?

`SystemNative_setattrlist`.

> And what #if could I use to conditionally compile it.

See other `check_symbol_exists` in `src/libraries/Native/Unix/configure.cmake`

Reply:
This file looks fine to me, I would just rename the Entry Points to fully align with [OSX/Interop.libobjc.cs](https://github.com/dotnet/runtime/blob/690126250a32df09cb09f9c93c0c83ad2bef9733/src/libraries/Common/src/Interop/OSX/Interop.libobjc.cs), but I'm no expert in this. 
cc @dotnet/platform-deps-team @ViktorHofer 
```suggestion
        [DllImport(Libraries.libc, EntryPoint = "libc_setattrlist", SetLastError = true)]
```

Reply:
> This file looks fine to me, I would just rename the Entry Point to fully align with [OSX/Interop.libobjc.cs](https://github.com/dotnet/runtime/blob/690126250a32df09cb09f9c93c0c83ad2bef9733/src/libraries/Common/src/Interop/OSX/Interop.libobjc.cs), but I'm no expert in this.

The `EntryPoint` is the name of the native method in the library (in this case `libc`). see https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportattribute.entrypoint?view=net-5.0. The reason that those methods are called `objc_...` is because that's what the native method is called. So as long as the library is `Libraries.libc` this won't work, but as @tmds was saying, we could make a System.Native shim, but this doesn't seem necessary to me since the method is only used in 1 spot.
Also, the guidelines say that shims are usually only used when similar apis exist across different unix platforms, but have different ABIs (https://github.com/dotnet/runtime/blob/462ab6e7a65dec97944e408b1959e71276f83988/docs/coding-guidelines/interop-guidelines.md#unix-shims), which is not the case for this since we're only using it on apple platforms and it is a common abi. In my opinion, we shouldn't make a System.Native shim because it would quite literally just be calling the native api (and do nothing else) and just double the code that needs to be maintained.
In summary, the `EntryPoint` should be left as is.

Reply:
> In summary, the EntryPoint should be left as is.

Yup, sounds good to me.

Reply:
@stephentoub should this use a shim, or is it fine as is?

Reply:
@stephentoub, I'm going to resolve this since it's almost ready for merging and I'm quite sure that it's fine as is, but feel free to let us know either way (and unresolve if not fine) :)
Thanks!